hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -729,7 +739,18 @@ private static Column createFilterCondition(Dataset dataset,String[] allColIdInS
                 continue; // Cannot Push Down Qualifier
             Column col = dataset.col(allColIdInSpark[q.getStoragePosition()]);
             q.clearOrderableCache();
-            Object value = (probeValue==null || i!=0?q.getOrderable():probeValue).getObject();
+
+            DataValueDescriptor dvd = probeValue;
+            if (dvd == null || i != 0) {
+                dvd = q.getOrderable();
+            }
+
+            if (dvd instanceof  SQLChar) {
+                col = adaptSQLCharColumn(col, (SQLChar) dvd);
```

Comment:
I set up parquet TPC-DS tables via instructions at https://splicemachine.atlassian.net/browse/DB-7852 but could not reproduce the problem reported with the following query:

select top 4 * from customer_demographics
where cd_education_status = 'Primary';

If there is an issue, then the fix looks fine.  But I could not reproduce it.  Are you doing something special in your test, like using the spark adapter?

Reply:
We can also test this with a smaller table, data must be populated from spark-sql:
```create table  ptba (col1 char(32))
  using parquet
  location '<some_path>/ptba';
insert into ptba values 
    ('abc'), 
    ('short string1');```
Then from splice sql, create the external table and query data:
```create external table  ptba (col1 char(32))
  stored as parquet
  location '<some_path>/ptba';
 select * from ptba where col1 = 'abc';```

Reply:
OK, this answers my question.  If the external table data is populated using Spark SQL as opposed to splice SQL, the CHAR column data is not padded with spaces, otherwise it is.  I wonder if the padding should be applied in a more common location when the dataset is first created.  We could also have a case with a join predicate between an external table and an internal table.  What happens in that case?  Since the external table column is not padded, and the internal table column is, would the join return no rows?

Reply:
I believe that currently we do not have good support mixing of external, internal tables, or external tables DDL after creation. But we do want to improve that. I am thinking of the same direction, i.e. to validate and adapt the tables at creation time. It will be at a larger scope. Let's discuss more on Jira then?

Reply:
OK, we can defer this to another Jira.  I was hoping you could do a quick test to see if it's an issue.  Then we could open another Jira for it.

Reply:
I agree with your @msirek, I think we should apply this transformation in SparkDataSetProcessor.processExternalDataset(), before calling createFilterCondition(), that way we don't have to do anything special with the filter and we'd handle joins etc correctly. There's too much stuff that can go wrong right now. 

Can you do that instead @ngpham ? After we get the dataset in processExternalDataset() we'd fix any SQLChar columns .